### ISSUE TITLE
EXC_BAD_ADDRESS supposedly on large strings

### ISSUE THREAD

**Issue by dsemenovsky** [LiquidPlayer/LiquidCore#145]
I get this error when processing large data.
The same code works perfectly on Android.

<img width="1521" alt="Screen Shot 2019-12-18 at 9 17 40 PM" src="https://user-images.githubusercontent.com/367111/71139493-4b6a4980-21dc-11ea-9996-596b08368cf3.png">

**Comment by ericwlange** [comment]
@dsemenovsky Thanks for reporting.  The issue is that I am allocating the temporary new string on the stack, and for large strings it blows the stack.  The solution is to allocate on the heap instead.  

For a quick fix, replace:
```c++
    char temp[length + 1];
```
with
```c++
    char *temp = (char*) malloc(length + 1);
```

and then after `JSStringRelease(s)`, put the following line:
```c++
    free(temp);
```

**Comment by dsemenovsky** [comment]
@ericwlange Is the best way of doing it to fork the repo and install using Carthage from a fork after making these changes?

**Comment by ericwlange** [comment]
accidental closure

**Comment by ericwlange** [comment]
This is fixed in 0.7.x
